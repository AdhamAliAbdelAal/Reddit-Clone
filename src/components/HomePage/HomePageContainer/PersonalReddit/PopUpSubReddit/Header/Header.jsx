import CloseIcon from '@mui/icons-material/Close';
import { IconButton } from '@mui/material';
import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
import {
  BOX, DialogTitle, Disc, Name, SecConatiner, StyledTooltip,
} from './style';
/**
 *@component
 * @param {object} props
 * @return {React.Component} -Header for pop up form
 */
function Header(props) {
  const { handleClose } = props;
  return (
    <>
      {' '}
      <BOX>
        <DialogTitle>Create a community</DialogTitle>
        <CloseIcon sx={{ cursor: 'pointer' }} onClick={handleClose} />
      </BOX>
      <SecConatiner>
        <Name>Name</Name>
        <Disc sx={{ height: 'fit-content', alignItems: 'center' }}>
          Community names including capitalization cannot be changed.
          <StyledTooltip
            title='Names cannot have spaces (e.g., "r/bookclub" not "r/book club"), must be between 3-21 characters, and underscores ("_") are the only special characters allowed. Avoid using solely trademarked names (e.g., "r/FansOfAcme" not "r/Acme").'
          >
            <IconButton>
              <ErrorOutlineIcon sx={{ transform: 'rotate(180deg)', fontSize: 'small', color: 'gray' }} />
            </IconButton>
          </StyledTooltip>
        </Disc>
      </SecConatiner>
    </>

  );
}
export default Header;
