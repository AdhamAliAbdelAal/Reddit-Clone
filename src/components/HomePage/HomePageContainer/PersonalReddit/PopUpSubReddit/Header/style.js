import {
  styled,
  Box, DialogContent, DialogTitle as DialogT, Tooltip, tooltipClasses,
} from '@mui/material';

export const BOX = styled(Box)({
  fontWeight: 500,
  fontSize: 16,
  color: '#1c1c1c',
  display: 'flex',
  // marginBottom: 16,
  padding: '0 0 16px 0px',
  justifyContent: 'space-between',
  borderBottom: '1px solid #edeff1',
});
export const DialogTitle = styled(DialogT)({
  fontWeight: 500,
  fontSize: 16,
  color: '#1c1c1c',
  display: 'flex',
  justifyContent: 'space-between',
  padding: 0,
});
export const SecConatiner = styled(DialogContent)({
  marginBottom: 8,
  display: 'flex',
  flexFlow: 'row wrap',
  marginRight: 8,
  maxWidth: '100%',
  flexDirection: 'column',
  paddingLeft: 0,
});
export const Name = styled('h3')({
  fontSize: 16,
  color: '#1c1c1c',
  display: 'block',
  marginBottom: 4,
  height: 20,
  marginTop: 0,
});

export const Disc = styled('p')({
  fontWeight: 400,
  color: '#7c7c7c',
  fontSize: 12,
  height: 16,
  margin: 0,
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'flex-start',
});
export const StyledTooltip = styled(({ className, ...props }) => (
  // eslint-disable-next-line react/jsx-props-no-spreading
  <Tooltip {...props} arrow classes={{ popper: className }} />
))(({ theme }) => ({
  [`& .${tooltipClasses.arrow}`]: {
    color: theme.palette.common.black,
  },
  [`& .${tooltipClasses.tooltip}`]: {
    backgroundColor: theme.palette.common.black,
  },
}));
