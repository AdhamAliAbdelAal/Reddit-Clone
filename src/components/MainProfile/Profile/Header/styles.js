import {
  AppBar, Box, Button, styled,
} from '@mui/material';

export const HeaderBox = styled(Box)(({ theme, subtitle }) => ({
  minHeight: 40,
  display: 'flex',
  // justifyContent: 'center',
  marginLeft: 24,
  ...((subtitle === undefined
|| subtitle === '?sort=new'
|| subtitle === '?sort=hot'
|| subtitle === '?sort=top'
|| subtitle === '?sort=top&t=day'
  ) && {
    marginLeft: 'calc((100% - 976px) / 2);',
  }),
  [theme.breakpoints.between('0', '970')]: {
    marginLeft: 24,
  },
  [theme.breakpoints.between('0', '630')]: {
    marginLeft: 5,
  },

}));

export const PageHeader = styled(AppBar)(() => ({
  backgroundColor: 'white',
  color: '#1a1a1b',
  boxShadow: 'none',

}));

export const HeaderButton = styled(Button)(({
  theme, condition, responsive, removedots,
}) => ({
  fontWeight: 500,
  fontSize: 14,
  fontFamily: "'IBM Plex Sans', 'sans-serif'",
  marginRight: 5,
  '&:hover': {
    backgroundColor: 'transparent',
    color: theme.palette.primary.main,
  },
  ...((condition === 'true') && {
    '&:before': {
      content: '" "',
      width: '100%',
      height: 2,
      backgroundColor: theme.palette.primary.main,
      position: 'absolute',
      bottom: 0,

    },
    color: theme.palette.primary.main,
  }),
  ...((responsive === 'res1') && {
    [theme.breakpoints.between('0', '1000')]: {
      display: 'none',
    },
  }),
  ...((responsive === 'res2') && {
    [theme.breakpoints.between('0', '900')]: {
      display: 'none',
    },
  }),
  ...((responsive === 'res3') && {
    [theme.breakpoints.between('0', '700')]: {
      display: 'none',
    },
  }),
  ...((responsive === 'res4') && {
    [theme.breakpoints.between('0', '500')]: {
      display: 'none',
    },
  }),
  ...((responsive === 'res5') && {
    [theme.breakpoints.between('0', '400')]: {
      display: 'none',
    },
  }),
  ...((responsive === 'res6') && {
    [theme.breakpoints.between('0', '350')]: {
      display: 'none',
    },
  }),
  ...((removedots === 'true') && {
    [theme.breakpoints.between('1000', '3000')]: {
      display: 'none',
    },
  }),
}));

export const SelectBox = styled(Box)(({ theme }) => ({
  maxWidth: 200,
  top: 30,
  border: ' 1px solid #eee',
  borderRadius: 4,
  boxShadow: '0 2px 4px 0 #eee',
  backgroundColor: 'white',
  position: 'absolute',
  zIndex: 10,
  [theme.breakpoints.between('0', '512')]: {
    marginLeft: -40,
  },
}));

export const SelectItem = styled(Button)(({ theme, responsive }) => ({
  padding: 5,
  color: 'black',
  fontFamily: "'IBM Plex Sans', 'sans-serif'",
  fontSize: 14,
  fontWeight: 500,
  width: '100%',
  justifyContent: 'left',
  textTransform: 'none',
  display: 'none',
  '& .MuiButtonBase-root:hover': {
    color: 'black',
    backgroundColor: '#d7d7d7',
  },
  ...((responsive === 'res1') && {
    [theme.breakpoints.between('0', '1000')]: {
      display: 'block',
    },
  }),
  ...((responsive === 'res2') && {
    [theme.breakpoints.between('0', '900')]: {
      display: 'block',
    },
  }),
  ...((responsive === 'res3') && {
    [theme.breakpoints.between('0', '700')]: {
      display: 'block',
    },
  }),
  ...((responsive === 'res4') && {
    [theme.breakpoints.between('0', '500')]: {
      display: 'block',
    },
  }),
  ...((responsive === 'res5') && {
    [theme.breakpoints.between('0', '400')]: {
      display: 'block',
    },
  }),
  ...((responsive === 'res6') && {
    [theme.breakpoints.between('0', '350')]: {
      display: 'block',
    },
  }),
}));
