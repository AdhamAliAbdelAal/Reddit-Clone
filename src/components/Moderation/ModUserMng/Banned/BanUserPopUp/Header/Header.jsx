import { Typography } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import StyledBox from './styles';

function Header(props) {
  const { handleClickCloseBan } = props;
  return (
    <StyledBox>
      <Typography>Ban a user:</Typography>
      <CloseIcon
        sx={{ cursor: 'pointer' }}
        onClick={handleClickCloseBan}
      />
    </StyledBox>
  );
}

export default Header;
