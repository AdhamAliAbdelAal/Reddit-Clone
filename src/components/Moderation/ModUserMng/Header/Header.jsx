import { Typography } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import StyledBox from './styles';

function Header(props) {
  const { buttonFunction, headerText } = props;
  return (
    <StyledBox>
      <Typography
        color="#1C1C1C"
        fontSize="16px"
        fontWeight={600}
      >
        {headerText}

      </Typography>
      <CloseIcon
        sx={{ cursor: 'pointer' }}
        onClick={buttonFunction}
      />
    </StyledBox>
  );
}

export default Header;
