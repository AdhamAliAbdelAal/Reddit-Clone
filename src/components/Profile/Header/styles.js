import {
  AppBar, Box, Button, styled,
} from '@mui/material';

export const HeaderBox = styled(Box)(() => ({
  minHeight: 40,
  display: 'flex',
  justifyContent: 'center',
}));

export const PageHeader = styled(AppBar)(() => ({
  backgroundColor: 'white',
  color: '#1a1a1b',
  boxShadow: 'none',

}));

export const HeaderButton = styled(Button)(({ theme, condition }) => ({
  fontWeight: 500,
  fontSize: 14,
  fontFamily: "'IBM Plex Sans', 'sans-serif'",
  marginRight: 5,
  '&:hover': {
    backgroundColor: 'transparent',
    color: theme.palette.primary.main,
  },
  ...((condition === 'true') && {
    '&:before': {
      content: '" "',
      width: '100%',
      height: 2,
      backgroundColor: theme.palette.primary.main,
      position: 'absolute',
      bottom: 0,

    },
    color: theme.palette.primary.main,
  }),
}));
