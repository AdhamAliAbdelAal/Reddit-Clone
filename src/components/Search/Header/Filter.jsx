import { useMemo } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

import {
  Header, HeaderContainer, MyCategory,
} from './style';

function useQuery() {
  const { search } = useLocation();

  return useMemo(() => new URLSearchParams(search), [search]);
}

/**
 * filter Search by its type
 *
 * @component FilterSearch
 * @property  {function} handleClick navigate

 * @returns {React.Component} filter Search by its type
 */
function FilterSearch() {
  const query = useQuery();
  const type = query.get('type') || 'posts';
  const navigate = useNavigate();

  // navigate
  const handleClick = (subPage) => {
    navigate(`?type=${subPage}`);
  };

  return (
    <HeaderContainer>
      <Header>
        <MyCategory
          condition={(type === 'posts').toString()}
          onClick={() => { handleClick('posts'); console.log((type === 'posts').toString()); }}
        >
          Posts
        </MyCategory>
        <MyCategory
          condition={(type === 'comments').toString()}
          onClick={() => { handleClick('comments'); }}
        >
          Comments
        </MyCategory>
        <MyCategory
          condition={(type === 'communities').toString()}
          onClick={() => { handleClick('communities'); }}
        >
          Communities
        </MyCategory>
        <MyCategory
          condition={(type === 'people').toString()}
          onClick={() => { handleClick('people'); }}
        >
          People
        </MyCategory>
      </Header>
    </HeaderContainer>
  );
}

export default FilterSearch;
