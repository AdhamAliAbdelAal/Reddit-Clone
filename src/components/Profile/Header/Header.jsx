import Box from '@mui/material/Box';
import { useNavigate, useParams } from 'react-router-dom';
import { PageHeader, HeaderBox, HeaderButton } from './styles';

function Header() {
  const { subTitle } = useParams();
  const navigate = useNavigate();

  // navigate
  const handleClick = (subPage) => {
    navigate(`${subPage}`);
    // console.log(filteration);
  };
  return (
    <Box sx={{ flexGrow: 1 }}>
      <PageHeader position="static">
        <HeaderBox>
          <HeaderButton
            color="inherit"
            onClick={() => { handleClick(''); }}
            condition={(subTitle === undefined || subTitle.includes('sort')).toString()}
          >
            overview

          </HeaderButton>
          {['posts', 'comments', 'history', 'saved', 'hidden', 'upvoted', 'downvoted'].map((text) => (
            <HeaderButton
              color="inherit"
              key={text}
              onClick={() => { handleClick(`${text}`); }}
              condition={(subTitle === text).toString()}
            >
              {text}

            </HeaderButton>
          ))}
        </HeaderBox>
      </PageHeader>
    </Box>
  );
}

export default Header;
