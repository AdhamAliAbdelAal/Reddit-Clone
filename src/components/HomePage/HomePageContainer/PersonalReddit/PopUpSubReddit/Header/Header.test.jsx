import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
