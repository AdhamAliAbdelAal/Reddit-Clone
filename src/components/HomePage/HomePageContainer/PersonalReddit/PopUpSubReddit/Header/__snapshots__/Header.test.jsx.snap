// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  " ",
  <div
    className="MuiBox-root css-1qcax3w"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-108k2dz-MuiTypography-root-MuiDialogTitle-root"
    >
      Create a community
    </h2>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw08pq-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </div>,
  <div
    className="MuiDialogContent-root css-qs439a-MuiDialogContent-root"
  >
    <h3
      className="css-118ivwj"
    >
      Name
    </h3>
    <p
      className="css-5o9jza"
    >
      Community names including capitalization cannot be changed.
      <button
        aria-label="Names cannot have spaces (e.g., \\"r/bookclub\\" not \\"r/book club\\"), must be between 3-21 characters, and underscores (\\"_\\") are the only special characters allowed. Avoid using solely trademarked names (e.g., \\"r/FansOfAcme\\" not \\"r/Acme\\")."
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1n93fd1-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </p>
  </div>,
]
`;
